### DISCUSSION TITLE
Restore optimization/scheduler states saved without pytorch-lightning

### DISCUSSION THREAD

**Discussion by ewrfcas** [Lightning-AI/pytorch-lightning#13698] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
If I have pre-trained a model (such as VIT) with its model and optimization state without saving by pytorch-lightning, how to restore the optimization for pytorch-lightning?
The model weights can be easily loaded by `load_state_dict`, but how to access `optimization` and further retore the state of it?

**Comment by ewrfcas** [comment]
Here is my solution, but I think it is not elegant.

```
model = LightningModule(...)
model.opt_state = None
model.opt_state = torch.load(ckpt_path)['opt']
```

Then for the `LightningModule` model:
```
def configure_optimizers(self):
      opt = Adam(...)
      if self.opt_state is not None:
            opt.load_state_dict(self.opt_state)
     return opt
```

I am looking for a better solution.
